pandas/core/indexes/multi.py

```diff
@@ -89,6 +89,9 @@ def __new__(cls, levels=None, labels=None, sortorder=None, names=None,
             raise TypeError("Must pass both levels and labels")
         if len(levels) != len(labels):
             raise ValueError('Length of levels and labels must be the same.')
+        if names is not None and len(names) > 0 and len(levels) == 0:
+            levels = [[] for _ in names]
+            labels = [[] for _ in names]
```

Comment:
yes, I believe you can remove those changes (and just let it pass thru here then)